MetaMask/metamask-extension
# feat: bypass add account confirmations for preinstalled snaps

## **Description**

This feature bypasses the standard add account confirmations flow for [PREINSTALLED_SNAPS](https://github.com/MetaMask/metamask-extension/blob/feat/preinstalled-snap-confirmations-bypass/app/scripts/snaps/preinstalled-snaps.ts#L4). Optionally, preinstalled snaps may set the `displayConfirmations` flag to `true` when they emit a `"notify: account created"` event, which will then enable them to show the standard add account confirmations.

#### Included
- Skips add account confirmation flow for preinstalled snaps
- Implements options for `displayConfirmation` and `accountNameSuggestion` in snap keyring builder

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24876?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/407

## **Manual testing steps**

**NO LONGER APPLICABLE** ❌ 
1. You will need to remove `!snap.preinstalled` from the filter in the `getSnapsList` selector [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/selectors/selectors.js#L2503)
2. Go to three vertical dot menu > Snaps > Account Watcher
3. Input an address or ENS (if on mainnet) into the form and click "Watch account"
4. You should **NOT** see any confirmations screens and see the selected account on the top of the extension change to your newly added address

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-extension/assets/91970214/8fc99966-182d-4068-970b-cea641ef4ab5

### **After**

https://github.com/MetaMask/metamask-extension/assets/91970214/8e1460fa-97bd-4a98-a654-53adbcf4032b

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
